librarian_packages/hera_librarian/tests/test_base_store.py

```diff
@@ -0,0 +1,183 @@
+
+    return
+
+
+def test_chmod(local_store):
+    # make a small test file on the store, then change permissions
+    local_store._ssh_slurp("touch /tmp/my_empty_file")
+    local_store._chmod("my_empty_file", "664")
+
+    # make sure permissions are correct
+    output = local_store._ssh_slurp("ls -l /tmp/my_empty_file")
+    perms = output.split(" ")[0]
+    assert perms == "-rw-rw-r--"
+
+    return
+
+
+def test_move(local_store):
+    # test moving a file
+    # first make sure that target file does not exist
+    local_store._ssh_slurp("rm -f /tmp/my_moved_file")
+    local_store._ssh_slurp("touch /tmp/my_empty_file")
+    local_store._move("my_empty_file", "my_moved_file")
+    assert local_store._ssh_slurp("ls /tmp/my_moved_file") == "/tmp/my_moved_file\n"
+
+    # test trying to overwrite a file that already exists
+    with pytest.raises(RPCError):
+        local_store._ssh_slurp("touch /tmp/my_empty_file")
+        local_store._move("my_empty_file", "my_moved_file")
+
+    # remove existing files; test using chmod
+    local_store._ssh_slurp("rm -f /tmp/my_empty_file /tmp/my_moved_file")
+    local_store._ssh_slurp("touch /tmp/my_empty_file")
+    local_store._move("my_empty_file", "my_moved_file", chmod_spec=664)
+    output = local_store._ssh_slurp("ls -l /tmp/my_moved_file")
+    perms = output.split(" ")[0]
+    assert perms == "-rw-rw-r--"
+
+    return
+
+
+def test_delete(local_store):
+    # test removing a file
+    local_store._ssh_slurp("touch /tmp/my_empty_file")
+    local_store._delete("my_empty_file")
+    assert (
+        local_store._ssh_slurp(
+            "if [ -f /tmp/my_empty_file ]; then echo file_still_exists; fi"
+        )
+        == ""
+    )
+
+    # test deleting a write-protected file
+    local_store._ssh_slurp("mkdir /tmp/my_empty_dir; chmod 755 /tmp/my_empty_dir")
+    local_store._delete("my_empty_dir", chmod_before=True)
+    assert (
+        local_store._ssh_slurp(
+            "if [ -d /tmp/my_empty_dir ]; then echo dir_still_exists; fi"
+        )
+        == ""
+    )
+
+    return
+
+
+def test_create_tempdir(local_store):
+    # make sure no temp dirs currently exist on host
+    local_store._ssh_slurp("rm -rf /tmp/libtmp.*")
+    tmppath = local_store._create_tempdir()
+    # we don't know exactly what the directory name will be, because a random
+    # 6-digit string is appended to the end
+    assert tmppath.startswith("libtmp.")
+    assert len(tmppath) == len("libtmp.") + 6
+    # make sure it exists on the host
+    assert (
+        local_store._ssh_slurp("ls -d1 /tmp/{}".format(tmppath))
+        == "/tmp/{}\n".format(tmppath)
+    )
+
+    return
+
+
+@ALL_FILES
+def test_get_info_for_path(local_store, datafiles):
+    # copy a datafile to store directory, so we can get its info
+    filepaths = list(map(str, datafiles.listdir()))
+    # make sure our files are ordered correctly
+    if filepaths[0].endswith("uvA"):
+        filepaths = filepaths[::-1]
```

Comment:
Hoo boy this seems super fragile and hacky. I feel like there's gotta be a better approach here ... `glob.glob`?

Reply:
Yeah... I've now put a `sorted()` around the `list()` call, so results are predictable.